### PR TITLE
Add deprecation note to Get-CsOnlineVoiceUser

### PR DESCRIPTION
**Pull request by JensTrier**
Adding deprecation note to Get-CsOnlineVoiceUser and description of alternative cmdlets
Update New/Set-CsCallingLineIdentity with more examples and information

### CODE REVIEW

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 1]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
 ```
 
 ## DESCRIPTION
+
+**Note**: This cmdlet will be deprecated. Please start using replacement cmdlets as soon as possible as described below.
+
+The table below lists the parameters to `Get-CsOnlineVoiceUser` and the alternative method of getting the same data using a combination of `Get-CsOnlineUser`, `Get-CsPhoneNumberAssignment`, `Get-CsOnlineLisLocation` and `Get-CsOnlineLisCivicAddress`
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:29]:
```suggestion
The following table lists the parameters to `Get-CsOnlineVoiceUser` and the alternative method of getting the same data using a combination of `Get-CsOnlineUser`, `Get-CsPhoneNumberAssignment`, `Get-CsOnlineLisLocation`, and `Get-CsOnlineLisCivicAddress`.
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 2]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+| :------------| :------- | :------- |
+| No parameters | Get information for all users | `Get-CsOnlineUser -Filter {(FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| CivicAddressId | Find phone number information where the assigned phone number is associcated to the CivicAddressId | `Get-CsPhoneNumberAssignment -CivicAddressId <CivicAddressId>` |
+| EnterpriseVoiceStatus | Find enabled users based on EnterpriseVoiceEnabled | `Get-CsOnlineUser -Filter {(EnterpriseVoiceEnabled -eq $True) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` or `Get-CsOnlineUser -Filter {(EnterpriseVoiceEnabled -eq $False) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User`  |
+| ExpandLocation | Show information about the LocationId | `Get-CsOnlineLisLocation -LocationId <LocationId>` |
+| Identity | Get information for a user | `Get-CsOnlineUser -Identity <Identity>` |
+| LocationId | Find phone number information where the assigned phone number is associated to the LocationId | `Get-CsPhoneNumberAssignment -LocationId <LocationId>` |
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:38]:
```suggestion
| LocationId | Find phone number information where the assigned phone number is associated with the LocationId | `Get-CsPhoneNumberAssignment -LocationId <LocationId>` |
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 3]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+| NumberAssigned | Find enabled users with a phone number assigned | `Get-CsOnlineUser -Filter {(LineUri -ne $Null) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| NumberNotAssigned | Find users without a phone number assigned | `Get-CsOnlineUser -Filter {(LineUri -eq $Null) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| PSTNConnectivity | Find enabled users with PhoneSystem (OnPremises) or CallingPlan (Online) | Online: `Get-CsOnlineUser -Filter {(FeatureTypes -contains 'CallingPlan') -and (AccountEnabled -eq $True)} -AccountType User` OnPremises: `Get-CsOnlineUser -Filter {-not (FeatureTypes -contains 'CallingPlan') -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:41]:
```suggestion
| PSTNConnectivity | Find enabled users with PhoneSystem (on-premises) or CallingPlan (online) | Online: `Get-CsOnlineUser -Filter {(FeatureTypes -contains 'CallingPlan') -and (AccountEnabled -eq $True)} -AccountType User` On-premises: `Get-CsOnlineUser -Filter {-not (FeatureTypes -contains 'CallingPlan') -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 4]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
 ## DESCRIPTION
+
+**Note**: This cmdlet will be deprecated. Please start using replacement cmdlets as soon as possible as described below.
+
+The table below lists the parameters to `Get-CsOnlineVoiceUser` and the alternative method of getting the same data using a combination of `Get-CsOnlineUser`, `Get-CsPhoneNumberAssignment`, `Get-CsOnlineLisLocation` and `Get-CsOnlineLisCivicAddress`
+
+| Parameter | Description | Alternative |
+| :------------| :------- | :------- |
+| No parameters | Get information for all users | `Get-CsOnlineUser -Filter {(FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| CivicAddressId | Find phone number information where the assigned phone number is associcated to the CivicAddressId | `Get-CsPhoneNumberAssignment -CivicAddressId <CivicAddressId>` |
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:34]:
```suggestion
| CivicAddressId | Find phone number information where the assigned phone number is associated with the CivicAddressId | `Get-CsPhoneNumberAssignment -CivicAddressId <CivicAddressId>` |
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 5]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+| Identity | Get information for a user | `Get-CsOnlineUser -Identity <Identity>` |
+| LocationId | Find phone number information where the assigned phone number is associated to the LocationId | `Get-CsPhoneNumberAssignment -LocationId <LocationId>` |
+| NumberAssigned | Find enabled users with a phone number assigned | `Get-CsOnlineUser -Filter {(LineUri -ne $Null) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| NumberNotAssigned | Find users without a phone number assigned | `Get-CsOnlineUser -Filter {(LineUri -eq $Null) -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+| PSTNConnectivity | Find enabled users with PhoneSystem (OnPremises) or CallingPlan (Online) | Online: `Get-CsOnlineUser -Filter {(FeatureTypes -contains 'CallingPlan') -and (AccountEnabled -eq $True)} -AccountType User` OnPremises: `Get-CsOnlineUser -Filter {-not (FeatureTypes -contains 'CallingPlan') -and (FeatureTypes -contains 'PhoneSystem') -and (AccountEnabled -eq $True)} -AccountType User` |
+
+Note: Get-CsOnlineUser filtering on FeatureTypes will be available later this year.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:43]:
```suggestion
**Note**: `Get-CsOnlineUser` filtering on feature types will be available later this year.
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 6]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+
+The table below lists the output fields from `Get-CsOnlineVoiceUser` and the alternative method of getting the same information using a combination of `Get-CsOnlineUser`, `Get-CsPhoneNumberAssignment` and `Get-CsOnlineLisLocation`.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:45]:
```suggestion
The following table lists the output fields from `Get-CsOnlineVoiceUser` and the alternative method of getting the same information using a combination of `Get-CsOnlineUser`, `Get-CsPhoneNumberAssignment`, and `Get-CsOnlineLisLocation`.
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 7]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+
+| Output field | Alternative |
+| :---------------------------------| :--------------------------------- |
+| Name | DisplayName in the output from `Get-CsOnlineUser` |
+| Id | Identity in the output from `Get-CsOnlineUser`|
+| SipDomain | Extract SipDomain from the SipAddress in output from `Get-CsOnlineUser` |
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:51]:
```suggestion
| Name | DisplayName in the output from `Get-CsOnlineUser` |
| Id | Identity in the output from `Get-CsOnlineUser`|
| SipDomain | Extract SipDomain from the SipAddress in the output from `Get-CsOnlineUser` |
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 8]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+| DataCenter | Extract DataCenter from RegistrarPool in the output from `Get-CsOnlineUser`|
+| TenantId | TenantId in the output from `Get-CsOnlineUser`|
+| PstnConnectivity | FeatureTypes in the output from `Get-CsOnlineUser`. If FeatureTypes contains CallingPlan, PstnConnectivity is 'Online'. If FeatureTypes contains 'PhoneSystem' and does not contain 'CallingPlan', PstnConnectivity is 'OnPremises' |
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:54]:
```suggestion
| DataCenter | Extract DataCenter from RegistrarPool in the output from `Get-CsOnlineUser`|
| TenantId | TenantId in the output from `Get-CsOnlineUser`|
| PstnConnectivity | FeatureTypes in the output from `Get-CsOnlineUser`. If FeatureTypes contains `CallingPlan`, PstnConnectivity is `Online`. If FeatureTypes contains `PhoneSystem` and does not contain `CallingPlan`, PstnConnectivity is `OnPremises` |
```

File: skype/skype-ps/skype/Get-CsOnlineVoiceUser.md [hunk 9]
```diff
@@ -23,6 +23,42 @@ Get-CsOnlineVoiceUser [-CivicAddressId <XdsCivicAddressId>] [-DomainController <
...
+| UsageLocation | UsageLocation in the output from `Get-CsOnlineUser` |
+| EnterpriseVoiceEnabled | EnterpriseVoiceEnabled in the output from `Get-CsOnlineUser` |
+| Number | LineUri in the output from `Get-CsOnlineUser`. You can get same phone number format by doing LineUri.Replace('tel:+','')  |
+| Location | Use LocationId in the output from `Get-CsPhoneNumberAssignment -AssignedPstnTargetId <Identity>` as the input to `Get-CsOnlineLisLocation -LocationId` |
+
+Note: Get-CsOnlineUser filtering on FeatureTypes will be available later this year.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineVoiceUser.md:60]:
```suggestion
**Note**: `Get-CsOnlineUser` filtering on FeatureTypes will be available later this year.
```

File: skype/skype-ps/skype/New-CsCallingLineIdentity.md [hunk 10]
```diff
@@ -63,6 +63,12 @@ New-CsCallingLineIdentity -Identity DKCQ -CallingIDSubstitute Resource -EnableUs
 
 This example creates a new Caller ID policy that sets the Caller ID to the phone number of the specified resource account and sets the Calling party name to Contoso
 
+### Example 5
+```
+New-CsCallingLineIdentity -Identity AllowAnonymousForUsers -EnableUserOverride $true
+```
+
+This example creates a new Caller ID policy that allows Teams user to make anonymous calls
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsCallingLineIdentity.md:71]:
```suggestion
This example creates a new Caller ID policy that allows Teams users to make anonymous calls.
```

File: skype/skype-ps/skype/New-CsCallingLineIdentity.md [hunk 11]
```diff
@@ -149,7 +155,11 @@ Accept wildcard characters: False
 ```
 
 ### -EnableUserOverride
-The EnableUserOverride switch lets the user override the Caller ID policy.
+The EnableUserOverride switch gives Microsoft Teams users the option under Settings and Calls to hide their phone number, when making outgoing calls. The CallerID will be Anonymous.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsCallingLineIdentity.md:158]:
```suggestion
The EnableUserOverride parameter gives Microsoft Teams users the option under Settings and Calls to hide their phone number when making outgoing calls. The CallerID will be Anonymous.
```

File: skype/skype-ps/skype/New-CsCallingLineIdentity.md [hunk 12]
```diff
@@ -149,7 +155,11 @@ Accept wildcard characters: False
...
+
+If CallingIDSubstitute is set to Anonymous, the EnableUserOverride has no effect, and the caller ID is always set to Anonymous.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsCallingLineIdentity.md:160]:
```suggestion
If CallingIDSubstitute is set to Anonymous, the EnableUserOverride parameter has no effect, and the caller ID is always set to Anonymous.
```

File: skype/skype-ps/skype/New-CsCallingLineIdentity.md [hunk 13]
```diff
@@ -149,7 +155,11 @@ Accept wildcard characters: False
...
+
+EnableUserOverride has precedence over other settings in the policy, unless substitution is set to Anonymous. For example, assume policy instance has substitution using a resource account and EnableUserOverride is set and enabled by the user. In this case, the outbound caller ID will be blocked and Anonymous will be used.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsCallingLineIdentity.md:162]:
```suggestion
EnableUserOverride has precedence over other settings in the policy unless substitution is set to Anonymous. For example, assume the policy instance has substitution using a resource account and EnableUserOverride is set and enabled by the user. In this case, the outbound caller ID will be blocked and Anonymous will be used.
```

File: skype/skype-ps/skype/Set-CsCallingLineIdentity.md [hunk 14]
```diff
@@ -66,6 +66,12 @@ Set-CsCallingLineIdentity -Identity DKCQ -CallingIDSubstitute Resource -Resource
 
 This example modifies the Caller ID policy that sets the Caller ID to the phone number of the specified resource account and sets the Calling party name to Contoso
 
+### Example 5
+```
+Set-CsCallingLineIdentity -Identity AllowAnonymousForUsers -EnableUserOverride $true
+```
+
+This example modifies the Caller ID policy and allows Teams user to make anonymous calls
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsCallingLineIdentity.md:74]:
```suggestion
This example modifies the Caller ID policy and allows Teams users to make anonymous calls.
```

File: skype/skype-ps/skype/Set-CsCallingLineIdentity.md [hunk 15]
```diff
@@ -139,7 +145,11 @@ Accept wildcard characters: False
 ```
 
 ### -EnableUserOverride
-The EnableUserOverride switch lets the user override the Caller ID policy.
+The EnableUserOverride switch gives Microsoft Teams users the option under Settings and Calls to hide their phone number, when making outgoing calls. The CallerID will be Anonymous.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsCallingLineIdentity.md:148]:
```suggestion
The EnableUserOverride parameter gives Microsoft Teams users the option under Settings and Calls to hide their phone number when making outgoing calls. The CallerID will be Anonymous.
```

File: skype/skype-ps/skype/Set-CsCallingLineIdentity.md [hunk 16]
```diff
@@ -139,7 +145,11 @@ Accept wildcard characters: False
...
+
+If CallingIDSubstitute is set to Anonymous, the EnableUserOverride has no effect, and the caller ID is always set to Anonymous.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsCallingLineIdentity.md:150]:
```suggestion
If CallingIDSubstitute is set to Anonymous, the EnableUserOverride parameter has no effect, and the caller ID is always set to Anonymous.
```

File: skype/skype-ps/skype/Set-CsCallingLineIdentity.md [hunk 17]
```diff
@@ -139,7 +145,11 @@ Accept wildcard characters: False
...
+
+EnableUserOverride has precedence over other settings in the policy, unless substitution is set to Anonymous. For example, assume policy instance has substitution using a resource account and EnableUserOverride is set and enabled by the user. In this case, the outbound caller ID will be blocked and Anonymous will be used.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsCallingLineIdentity.md:152]:
```suggestion
EnableUserOverride has precedence over other settings in the policy unless substitution is set to Anonymous. For example, assume the policy instance has substitution using a resource account and EnableUserOverride is set and enabled by the user. In this case, the outbound caller ID will be blocked and Anonymous will be used.
```

**Comment by JensTrier** [comment]
> Copy review - @JensTrier, kindly commit the suggested changes. Many thanks.

Thanks @JohanFreelancer9!! Done.

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by yogkumgit** [comment]
@dariomws @tseward @get-itips

**Review by dariomws — APPROVED** [review]

**Review by get-itips — APPROVED** [review]

**Review by tseward — APPROVED** [review]

**Comment by yogkumgit** [comment]
@tiburd Please merge this Pr. Thanks

**Pull request merged**